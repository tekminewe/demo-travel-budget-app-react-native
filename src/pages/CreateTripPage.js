// @flow
import React from 'react';
import {
  View,
  TextInput,
  Button,
  StyleSheet,
} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    marginTop: 30,
  }
});

const CreateTripPage = () => {
  return (
    <View style={styles.container}>
      <TextInput
        placeholder="Trip Name"
      />
      <Button
        title="Add"
        onPress={() => {}}
      />
    </View>
  );
}

CreateTripPage.navigationOptions = {
  title: 'New Trip',
}

export default CreateTripPage;
