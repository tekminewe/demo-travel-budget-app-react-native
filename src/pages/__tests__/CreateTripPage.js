import React from 'react';
import renderer from 'react-test-renderer';
import CreateTripPage from '../CreateTripPage';

describe('CreateTripPage', () => {
  test('should render form', () => {
    const page = renderer.create(<CreateTripPage />).toJSON();
    expect(page).toMatchSnapshot();
  });
});
