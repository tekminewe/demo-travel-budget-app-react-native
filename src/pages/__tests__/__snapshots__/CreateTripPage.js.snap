// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTripPage should render form 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "marginTop": 30,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    placeholder="Trip Name"
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
  />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add
      </Text>
    </View>
  </View>
</View>
`;
